lib/cli/test/unit/Cardano/CLISpec.hs

```diff
@@ -46,9 +57,18 @@ import qualified Data.Text.IO as TIO
 spec :: Spec
 spec = do
     describe "Can perform roundtrip textual encoding & decoding" $ do
+        textRoundtrip $ Proxy @Iso8601Time
         textRoundtrip $ Proxy @(Port "test")
         textRoundtrip $ Proxy @MnemonicSize
 
+    describe "Can decode basic ISO 8601 time examples" $ do
+        canDecodeIso8601Time "2008-09-15T15:53:00Z"
+        canDecodeIso8601Time "2008-09-15T15:53:00.1Z"
+        canDecodeIso8601Time "2008-09-15T15:53:00.12Z"
+        canDecodeIso8601Time "2008-09-15T15:53:00+05:00"
+        canDecodeIso8601Time "2008-09-15T15:53:00.1+05:00"
+        canDecodeIso8601Time "2008-09-15T15:53:00.12+05:00"
```

Comment:
I believe the `-` makes this "extended ISO8601", which the **API** shouldn't support according to #466 (modulo I don't know why it says union signs):

> note: We do not accept and ISO-8601 extended format (with union signs) because it conflicts with the range separator

Regardless what the CLI does, maybe some more test-cases could be good
- `"20080915T15:53:00.1+05:00"`
- `""`
?

Reply:
> (modulo I don't know why it says union signs):

Probably some French jargon that doesn't translate well in English :grimacing: . I've changed _union_ for `-` to make it clearer.

Reply:
Pointer: https://hackage.haskell.org/package/time-1.9.2/docs/Data-Time-Format-ISO8601.html#t:FormatExtension

Reply:
>  French jargon that doesn't translate well in English

I suspected that 😄

Reply:
> Regardless what the CLI does, maybe some more test-cases could be good

Thanks for looking at this. I've added a bunch more test cases, including some negative test cases.

Reply:
> I believe the `-` makes this "extended ISO8601", which the **API** shouldn't support according to #466 (modulo I don't know why it says union signs):

That's an excellent point, one that I hadn't considered -- thanks for raising it.

We had a discussion about this in the slack channel. In the end, it was decided to allow the extended format in the CLI, as this format is easier to parse visually.